custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py

```diff
@@ -216,9 +226,16 @@ def updates(self):
                 CASE WHEN last_referral_date BETWEEN %(start_date)s AND %(end_date)s
                 THEN 1 ELSE 0 END
             ) as cases_person_referred
-        FROM "{ucr_tablename}"
-        WHERE (opened_on <= %(end_date)s AND (closed_on IS NULL OR closed_on >= %(start_date)s ))
-        GROUP BY awc_id, supervisor_id) ut
+        FROM "{ucr_tablename}" ucr LEFT JOIN
+             "{adolescent_girls_table}" adolescent_girls_table ON
```

Comment:
Why is this using a JOIN instead of a separate UPDATE (which looks like is more common in `agg_awc`)?

Reply:
Hey, I Left joined `adolescent_girls_table` with this `person`  query because it will automatically will not consider the closed person cases automatically from `adolescent_girls_table`. I could have make it separate query but then I had to write something again to eliminate those.

Reply:
Ah. I'm not very familiar with the adolescent girl case. Does this mean that the girl case could be open when the person case is closed? If that's the case then this makes sense. Otherwise you could also check the closed status on the ag case

Reply:
Hey, yeah, So `adolescent_girls_table` is meant to contain data just for `adolescent girl AWC registration form` not the case itself([here](https://github.com/dimagi/commcare-hq/blob/172c2fbe60356b4d0bfb463cb588bf3569e9f8c2/custom/icds_reports/utils/aggregation_helpers/distributed/adolescent_girls_registration.py#L32)) Similar to thr_child table. Therefore this table alone can't determine whether for all cases for whom the form is submitted till date are open or closed. Therefore it needs the join here with person case. Along with just open/close thing, joining is giving `is_migrated` property of person case which is needed for this implementation. I am using the person_case Id which comes from the form itself.

On the other hand, we are not using `ag_care` case explicitly for this implementation because none of the property specific to ag_care case is needed for this report that's why I have not created another case UCR for `ag_care` case.

Reply:
Ah. yeah that make sense. For some reason I thought that this was based on the case data source